sys/socket.h is required by struct ucred.
stddef.h is required by offsetof().

--- src/journal/journald.h.orig	2012-08-26 09:16:47.178872884 +0200
+++ src/journal/journald.h	2012-08-26 09:17:00.879545440 +0200
@@ -23,7 +23,9 @@
 
 #include <inttypes.h>
 #include <sys/types.h>
+#include <sys/socket.h>
 #include <stdbool.h>
+#include <stddef.h>
 
 #include "journal-file.h"
 #include "hashmap.h"
